spec.html

```diff
@@ -7308,7 +7308,7 @@ <h1>[[Call]] ( _thisArgument_, _argumentsList_ )</h1>
       <p>The [[Call]] internal method for an ECMAScript function object _F_ is called with parameters _thisArgument_ and _argumentsList_, a List of ECMAScript language values. The following steps are taken:</p>
       <emu-alg>
         1. Assert: _F_ is an ECMAScript function object.
-        1. If _F_.[[FunctionKind]] is `"classConstructor"`, throw a *TypeError* exception.
+        1. If _F_.[[IsClassConstructor]] is *true*, throw a *TypeError* exception.
```

Comment:
i wonder if maybe a better name is something that describes its purpose (throwing on [[Call]]) rather than the current category of things that cause it to be true?

something like “ThrowsOnCall” or something?

Reply:
Yup, good idea. Would `[[CallThrowsTypeError]]` be better, or too specific?

It's interesting that, instead of having this slot, `MakeClassConstructor` could just set the function's `[[Call]]` to `Throw a *TypeError* exception`. But that would make it an exotic function, which is undesirable, I guess.

Reply:
The effect is the same imo, since it behaves differently than a normal one ¯\\\_(ツ)_/¯  but probably best to stick to the slot approach.

Your name suggestion is fine too - just seems better to have something that speaks to the effect it has rather than the characteristic that leads to it having the effect.

Reply:
> The effect is the same imo,

Well, the effect of [[Call]] would be the same, but it's possible that changing a class constructor function from ordinary to exotic might have unintended side-effects (e.g. failed assertions).

> probably best to stick to the slot approach.

Yup.